picoscope/ps6000a.py

```
@@ -0,0 +1,963 @@
+
+    "Updates"
+
+    def _lowLevelCheckForUpdate(self):
+        """
+        Check whether a firmware update for the device is available.
+
+        Returns
+        -------
+        firmwareInfos : None
+            not implemented: A struct with information.
+        number : int
+            Number of elements in the structure.
+        required : bool
+            Whether an update is required or not.
+        """
+        # TODO what to do with the struct?
+        number = c_int16()
+        required = c_uint16()
+        m = self.lib.ps6000aCheckForUpdate(c_int16(self.handle), c_void_p(),
+                                           byref(number), byref(required))
+        self.checkResult(m)
+        return None, number, required
+
+    def _lowLevelStartFirmwareUpdate(self, function):
+        callback = updateFirmwareProgress(function)
+        m = self.lib.ps6000aStartFirmwareUpdate(c_int16(self.handle), callback)
+        self.checkResult(m)
+
+    "Misc"
+
+    ##################################
+    # TODO verify functions below here in the manual
+    # TODO ensure all relevant functions are in here
+    ##################################
+
+    "Measurement"
+
+    # Complicated triggering
+    # need to understand structs for some of this to work
+    def _lowLevelGetValuesTriggerTimeOffsetBulk():
+        pass
+
+    def _lowLevelSetTriggerChannelConditions():
+        pass
+
+    def _lowLevelSetTriggerChannelDirections():
+        pass
+
+    def _lowLevelSetTriggerChannelProperties():
+        pass
+
+    def _lowLevelSetTriggerDelay():
+        pass
+
+    def _lowLevelSetTriggerDigitalPortProperties(self):
+        pass
+
+    # Optional input triggering: PulseWidthQualifier
+    def _lowLevelSetPulseWidthQualifierConditions(self):
+        pass
+
+    def _lowLevelSetPulseWidthQualifierDirections(self):
+        pass
+
+    def _lowLevelSetPulseWidthQualifierProperties(self):
+        pass
```

It will be best, when we are done, to simply remove these functions so they cause a visible to the end user as opposed to silently failing.